tailcfg/tailcfg.go

```diff
@@ -830,6 +832,26 @@ type Hostinfo struct {
 	//       require changes to Hostinfo.Equal.
 }
 
+// VIPService represents a service created on a tailnet from the
+// perspective of a node providing that service. These services
+// have an virtual IP (VIP) address pair distinct from the node's IPs.
+type VIPService struct {
+	// Name is the name of the service, of the form `svc:dns-label`.
+	// See CheckServiceName for a validation func.
+	// Name uniquely identifies a service on a particular tailnet,
+	// and so also corresponds uniquely to the pair of IP addresses
+	// belonging to the VIP service.
+	Name string
+
+	// Ports specify which ProtoPorts are made available by this node
+	// on the service's IPs.
+	Ports []ProtoPortRange
```

Comment:
Question: Is this Ports the service config ports on this specific node or it's the general open ports for the VIP associated with the service?

Reply:
These are the ports that are locally configured for this service. The ports generally required for a service to function are in control

Reply:
> locally configured for this service. The ports generally required

(I might have missed this in a design or something) Could we clarify the distinction?

For context, for kube we assuming that it will be possible to advertise <vip-service-foo>:443, <vip-service-bar>:443 on the same node and that any backend addresses + ports will be solely in serve config only.
I was not assuming that there would be any other ports involved

Reply:
So when you create a service on control with your API endpoint, you provide a list of ports that any node would need to serve in order to advertise a service. These are the "generally required" ports I was talking about.

Then you add a mapping from `443` to whatever you want in the serve config. These are the "locally configured" ports I was talking about. Control only allows connections to flow to a node once everything in that list from step 1 is in the serve config. It doesn't matter what's in the serve config, as long as there's something in it that handles all the required ports

> For context, for kube we assuming that it will be possible to advertise :443, :443 on the same node and that any backend addresses + ports will be solely in serve config only.

Like, `serviceA:443` and `serviceB:443` from the same node? Yes, that should be possible. It's sort of orthogonal to this because this struct refers to only one service, we'll have an array of these structs to deal with multiple services

Reply:
ok, so TL;DR is that 'generally required' and 'locally configured' is the same port, but in one case configured via control and in the other case locally, thanks for the explanation 👍🏼 

> Control only allows connections to flow to a node once everything in that list from step 1 is in the serve config.

As a side note, I think this model might make it a bit difficult to use the VIP services in other contexts, i.e when we want to just let the node receive traffic on service VIP and decide what to do with it locally (i.e the kube operator L3 proxies that route to backend by local iptables rules or equivalent third party/user setups). But that is a discussion out of scope for this PR